src/GeositeFramework/js/Export.js

```diff
@@ -54,63 +68,79 @@ require(['use!Geosite'],
+                                .filter(function () { return this.checked; })
+                                .map(function () { return this.value; })
+                                .first());
                             if (pageOrientation === "Landscape") {
                                 $('<link>', {
                                     rel: 'stylesheet',
                                     href: 'css/print-landscape.css',
                                     'class': '.print-orientation-css',
                                 }).appendTo('head');
-                                _.delay(orientDeferred.resolve, 750);
-                                $(".legend-layer").each(function(el) { $(this).children(".item").wrapAll("<div class='print-legend-coll'></div>"); });
+                                _.delay(orientDeferred.resolve, 500);
                             } else {
                                 $('<link>', {
                                     rel: 'stylesheet',
                                     href: 'css/print-portrait.css',
                                     'class': '.print-orientation-css',
                                 }).appendTo('head');
-                                _.delay(orientDeferred.resolve, 750);
+                                _.delay(orientDeferred.resolve, 500);
                             }
 
                             $.when(orientDeferred).then(function () {
-                                window.print();
-                                previewDeferred.resolve();
+                                map.width = parseFloat($("#export-print-preview-map").css("width"));
+                                map.height = parseFloat($("#export-print-preview-map").css("height"));
+                                map.resize(true);
+                                map.centerAt(mapDimensions.extent.getCenter());
+                                _.delay(resizeDeferred.resolve, 500);
+
+                                // the legend items are affected by the map resize, so
+                                // they are manipulated just before printing
+                                $.when(resizeDeferred).then(function() {
+                                    // expand legend container if minimized
+                                    if ($("#legend-container-0").hasClass("minimized")) {
+                                        _.head($(".legend-close")).click();
+                                        postPrintAction = function() {
+                                            _.head($(".legend-close")).click();
+                                        };
+                                    }
+
+                                    if ($("[name='export-include-legend']").is(":checked")) {
+                                        // show & expand all legend items
+                                        $("#legend-container-0").css({ visibility: "visible" });
+                                        _.each(
```

Comment:
Here, too - jQuery will execute on collections be default.  This `_.each` can be replaced with:

```js
$(".item.expand>.expand-legend").click()
```